src/common.jl

```diff
@@ -0,0 +1,36 @@
+
+"""
+    MessageHandled(::Bool)
+
+`MessageHandled(false)` should be returned from a `handle_message` on a logger,
+if it did not actually handle the log message.
+For example, if the log message was below the level it should log.
+This is of particular relevance to the [`ActiveFilteredLogger`](@ref), which can't know
+til `handle_message` if a log message will be filtered or not.
+Ideally, `MessageHandled(true)` would be returned from loggers when when they
+successfully handled a message, however this is not strictly required.
+E.g Sinks should always return `MessageHandled(true)`.
+"""
+struct MessageHandled
+    val :: Bool
+end
+
+was_handled(response) = true  # If we don;t get a MessageHandled then assume worked
```

Comment:
I wonder if it makes sense to return `missing` here so that tee logger would use three-valued logic automatically.  The caller can then distinguish "definitely handled", "maybe handled", "not handled."  Example use-case may be that, in `FirstMatchLogger` #23:

* `!== false` is considered `true` for info level and below
* but only `== true` is considered `true` for more urgent messages (to make sure they are shown)

Not sure it's worth the complexity, though.

Reply:
Oooh interesting idea. Worth consideration

Reply:
Or could it make sense to just return an `Enum` value? Then we have the option to expand the possible statuses later if necessary. On the downside making that forward compatible could be annoying.

If people always write utility routing tools in terms of `was_handled` (or other verbs we provide) forward compatibility would be easier. Can we somehow arrange `was_handled` to be the main API with the type `MessageHandled` as secondary?

Reply:
Conversely: maybe `was_handled` should just be moved into `TeeLogger`
Right now it is only used to Booleanize the status so that can be written using `handled |= was_handled(status)`